### PR TITLE
Add sub-sector selection in report structure of export page

### LINKED ISSUE #1225: Export: Add 2 options in report structure to support sub-sectors
Support Row>Column and Column>Row with sub-sectors

### PR DESCRIPTION
**Pull request by samshara**
fixes https://github.com/the-deep/client/issues/1225
depends on https://github.com/the-deep/server/pull/620

### CODE REVIEW

**Review by AdityaKhatri — APPROVED** [review]

File: src/utils/framework.ts [hunk 1]
```diff
@@ -22,34 +22,36 @@ import {
...
+                    return ({
+                        id: `${s.id}-${d.id}-${sd.id}`,
+                        title: s.title,
+                        sublevels: subSectorLevels,
+                    });
+                }
+                return ({
+                    id: `${s.id}-${d.id}-${sd.id}`,
+                    title: s.title,
+                });
```
Comment (tnagorra) [inline src/utils/framework.ts]:
```suggestion
                return ({
                    id: `${s.id}-${d.id}-${sd.id}`,
                    title: s.title,
                    sublevels: includeSubSector
                        ? subsectors?.map(ss => ({
                            id: `${s.id}-${ss.id}-${d.id}-${sd.id}`,
                            title: ss.title,
                        }))
                        : undefined,
                });
```

Comment (tnagorra) [reply src/utils/framework.ts]: It's better to do things once

Comment (tnagorra) [reply src/utils/framework.ts]: In this case, setting the id for sector.

File: src/utils/framework.ts [hunk 2]
```diff
@@ -67,6 +69,58 @@ const transformMatrix2dLevels = ({
...
+                    id: `${s.id}-${ss.id}`,
+                    title: ss.title,
+                    sublevels: dimensions,
+                });
+            });
+        } else {
+            sublevels = widgetDim.map((d) => {
+                const { subdimensions } = d;
+                const subDimensionsLevel = subdimensions.map(sd => ({
+                    id: `${s.id}-${d.id}-${sd.id}`,
```
Comment (tnagorra) [inline src/utils/framework.ts]: Maybe we should create a simple function to do this repeated logic

```typescript
function createId<T extends string | number>(...args: [T, T, ...T[]]) {
    return args.join('-');
}
```

Comment (tnagorra) [reply src/utils/framework.ts]: This function will not accept null, undefined, or arguments with length less than 2.

File: src/views/Export/index.tsx [hunk 3]
```diff
@@ -234,9 +235,13 @@ function Export(props: Props) {
...
     useEffect(() => {
-        const structure = createReportStructure(analysisFramework, reportStructureVariant);
+        const structure = createReportStructure(
+            analysisFramework,
+            reportStructureVariant,
+            includeSubSector,
+        );
         setReportStructure(structure);
-    }, [analysisFramework, reportStructureVariant]);
+    }, [analysisFramework, reportStructureVariant, includeSubSector]);
```
Comment (tnagorra) [inline src/views/Export/index.tsx:244]: Can't we just use a useMemo for `reportStructure` value?

**Pull request merged**